pkg/container/docker_run.go

```diff
@@ -579,8 +578,8 @@ func (cr *containerReference) copyDir(dstPath string, srcPath string, useGitIgno
 			return err
 		}
 		log.Debugf("Writing tarball %s from %s", tarFile.Name(), srcPath)
-		defer tarFile.Close()
 		defer os.Remove(tarFile.Name())
+		defer tarFile.Close()
```

Comment:
We need to close the tarFile before removing it, since we use defer here the order were wrong before my change

Reply:
I think it would be more readable to create a defered inline func like this:

```go
defer (func () {
  tarFile.Close()
  os.Remove(tarFile.Name())
})()
```

Reply:
```suggestion
		defer func(tarFile *os.File) {
			name := tarFile.Name()
			err := tarFile.Close()
			if err != nil {
				logger.Error(err)
			}
			err = os.Remove(name)
			if err != nil {
				logger.Error(err)
			}
		}(tarFile)
```

Reply:
Thanks, printing the error is better. I was confused due to a lot of non deleted tmp files created by act, because windows keeps them for a long time.